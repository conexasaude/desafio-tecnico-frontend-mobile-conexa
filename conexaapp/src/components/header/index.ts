import {Button} from './button/view';
import {Greetings} from './greetings/view';
import {HStack} from './horizontal-stack/view';
import {Root} from './root/view';
import {Title} from './title/view';
import {VStack} from './vertical-stack/view';

export const Header = {
  Root: Root,
  Button: Button,
  Greetings: Greetings,
  VStack: VStack,
  HStack: HStack,
  Title: Title,
};
