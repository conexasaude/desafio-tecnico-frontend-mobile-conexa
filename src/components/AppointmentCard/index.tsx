import { GestureResponderEvent, Pressable } from 'react-native';
import { Container, Pacient, AppointmentDate, Field, Title, Header } from './styles';
import { MaterialCommunityIcons } from '@expo/vector-icons'; 
import formatsDate from '../../utils/formatsDate';

interface AppointmentCardProps {
    id: string
    onPress: (event: GestureResponderEvent) => void
    pacient: string
    date: string
    observation: string
}

export default function AppointmentCard({ id, date, observation, onPress, pacient }: AppointmentCardProps) {

  return (
    <Container id={id} onPress={onPress}>
        <Header>
            <Title>Consulta #{id}</Title>
            <AppointmentDate>{formatsDate(date, 'day-month-year')}</AppointmentDate>
        </Header>
        <Field>
            <MaterialCommunityIcons name="account" size={20} color="#FFF" />
            <Pacient>{pacient}</Pacient>
        </Field>
        <Field>
            <MaterialCommunityIcons name="clock" size={20} color="#FFF" />
            <AppointmentDate>{formatsDate(date, 'time')}</AppointmentDate>
        </Field>
    </Container>
  );
}
