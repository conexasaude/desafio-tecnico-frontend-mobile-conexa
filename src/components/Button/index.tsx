import { ButtonContainer, ButtonText } from './styles';

interface ButtonProps {
  onPress: () => Promise<void>
  label: string
  disabled?: boolean
  testID?: string
}

export default function Login({ label, onPress, disabled, testID }: ButtonProps) {

  return (
    <ButtonContainer testID={testID} onPress={onPress} disabled={disabled}>
      <ButtonText>{label}</ButtonText>
    </ButtonContainer>
  );
}