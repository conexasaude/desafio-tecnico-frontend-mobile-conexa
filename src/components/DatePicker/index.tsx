import { GestureResponderEvent, Pressable, Text } from 'react-native';
import { Label, Placeholder, Value, ValueContainer } from './styles';

interface DatePickerProps {
    label: string
    value: Date | null
    onPress: ((event: GestureResponderEvent) => void) | null | undefined
}

export default function DatePicker({ label, value, onPress }: DatePickerProps) {
    function valueFormat(date: Date) {
        const formattedDate = new Intl.DateTimeFormat('pt-BR', {
			year: 'numeric',
			month: '2-digit',
			day: '2-digit',
			hour: '2-digit',
			minute: '2-digit',
		}).format(date);

        return formattedDate
    }
    return (
        <Pressable onPress={onPress}>
            <Label>
                {label}
                <Text style={{ color: 'red' }}> *</Text>
            </Label>
            
            <ValueContainer>
                {value ? <Value>{valueFormat(value)}</Value> : <Placeholder>Selecione uma data</Placeholder> }
            </ValueContainer>
        </Pressable>
    )
}