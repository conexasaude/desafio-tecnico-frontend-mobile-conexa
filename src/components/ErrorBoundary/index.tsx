import React from 'react';
import { useTheme } from '@emotion/react';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import * as Styled from './styles';

interface ErrorBoundaryProps {
  tryAgain: () => void;
}

export function ErrorBoundary({ tryAgain }: ErrorBoundaryProps) {
  const { colors } = useTheme();

  return (
    <Styled.Container testID="error-boundary">
      <Icon
        name="alert-circle-outline"
        color={colors.status.warning}
        size={64}
      />
      <Styled.Text>
        {'Desculpe, ocorreu um problema. Por favor, tente novamente.'}
      </Styled.Text>

      <Styled.ButtonTryAgain onPress={tryAgain} title="Tentar Novamente" />
    </Styled.Container>
  );
}
