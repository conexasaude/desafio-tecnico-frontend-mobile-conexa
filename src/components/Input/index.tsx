import React, { forwardRef } from 'react'

import { TextInput as RNTextInput, TextInputProps } from 'react-native'
import {
  Container,
  Label,
  TextInput,
  Error,
  ContainerRow,
  Icon,
} from './styles'
import { useTheme } from '@emotion/react'

import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons'

interface InputProps extends TextInputProps {
  label: string
  error?: string
  icon?: string
  onPressIcon?: () => void
}
export const Input = forwardRef<RNTextInput, InputProps>(
  ({ label, error, icon, onPressIcon, ...rest }, ref) => {
    const { COLORS } = useTheme()

    return (
      <Container>
        <Label error={!!error}>{label}</Label>

        <ContainerRow>
          <TextInput
            ref={ref}
            placeholderTextColor={COLORS.PHILIPINE_GRAY}
            error={!!error}
            {...rest}
          />

          {icon && (
            <Icon>
              <MaterialCommunityIcons
                name={icon}
                onPress={onPressIcon}
                size={24}
              />
            </Icon>
          )}
        </ContainerRow>

        {error && <Error>{error}</Error>}
      </Container>
    )
  },
)

Input.displayName = 'Input'
