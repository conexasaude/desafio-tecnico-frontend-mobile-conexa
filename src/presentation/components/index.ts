export * from './buttons';
export * from './heading';
export * from './icons';
export * from './inputs';
