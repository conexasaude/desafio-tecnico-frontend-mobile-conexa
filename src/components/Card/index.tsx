import React from 'react';
import * as Styled from './styles';
import { TouchableOpacityProps } from 'react-native';
import { ButtonProps } from '@/components/Button';

interface CardProps extends TouchableOpacityProps {
  title: string;
  description: string;
  button: ButtonProps;
}

export function Card({ title, description, button, ...rest }: CardProps) {
  return (
    <Styled.Wrapper
      testID="card"
      activeOpacity={0.8}
      onPress={button.onPress}
      {...rest}>
      <Styled.Row>
        <Styled.Column>
          <Styled.Title numberOfLines={1}>{title}</Styled.Title>
          <Styled.Description numberOfLines={2}>
            {description}
          </Styled.Description>
        </Styled.Column>

        <Styled.SeeMore
          testID="card-button"
          variant="outlined"
          activeOpacity={0.4}
          {...button}
        />
      </Styled.Row>
    </Styled.Wrapper>
  );
}
