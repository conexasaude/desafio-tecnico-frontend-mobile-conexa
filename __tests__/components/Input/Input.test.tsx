import React from 'react';
import {
  renderWithThemeProvider,
  matchSnapshotWithProvider,
} from '@/lib/functions/Testing';

import { Input } from '@/components/Input';
import { fireEvent } from '@testing-library/react-native';

describe('Input', () => {
  test('should render correctly', () => {
    renderWithThemeProvider(<Input label="label" />);
  });

  test('should matches snapshot', () => {
    matchSnapshotWithProvider(<Input label="label" />);
  });

  test('should render label correctly', () => {
    const { getByText } = renderWithThemeProvider(<Input label="label" />);
    expect(getByText('label')).toBeDefined();
  });

  test('should render icon visibility when secureTextEntry is true', () => {
    const { getByTestId } = renderWithThemeProvider(
      <Input label="label" secureTextEntry />,
    );

    expect(getByTestId('visibility-icon')).toBeDefined();
  });

  test('should render icon visibility-off when press button visibilty and return to visibiliy when press again', () => {
    const { getByTestId } = renderWithThemeProvider(
      <Input label="label" secureTextEntry />,
    );

    const input = getByTestId('input-text');
    const visibilityIcon = getByTestId('visibility-icon');

    expect(visibilityIcon).toBeTruthy();
    expect(input.props.secureTextEntry).toBeTruthy();

    fireEvent.press(visibilityIcon);

    const visibilityOffIcon = getByTestId('visibility-off-icon');
    expect(visibilityOffIcon).toBeTruthy();
    expect(input.props.secureTextEntry).toBeFalsy();

    fireEvent.press(visibilityOffIcon);

    expect(visibilityIcon).toBeTruthy();
    expect(input.props.secureTextEntry).toBeTruthy();
  });

  test('should render error text', () => {
    const { getByText } = renderWithThemeProvider(
      <Input label="label" error="something went wrong" />,
    );
    expect(getByText('something went wrong')).toBeDefined();
  });

  test('should render a component textarea', () => {
    const { getByTestId } = renderWithThemeProvider(
      <Input label="label" textArea />,
    );

    const containerInput = getByTestId('input-container');
    const input = getByTestId('input-text');

    expect(containerInput.props.style[0].height).toEqual(180);
    expect(input.props.multiline).toBeTruthy();
    expect(input.props.numberOfLines).toEqual(6);
    expect(input.props.textAlignVertical).toEqual('top');
  });
});
