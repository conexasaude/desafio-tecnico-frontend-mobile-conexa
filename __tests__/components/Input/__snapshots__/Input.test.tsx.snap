// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should matches snapshot 1`] = `
<View
  style={
    [
      {},
      {},
    ]
  }
>
  <View
    style={
      [
        {
          "borderColor": "#2E50D4",
          "borderRadius": 8,
          "borderWidth": 2,
          "height": 52,
          "marginTop": 16,
          "padding": 8,
        },
        undefined,
      ]
    }
    testID="input-container"
    textArea={false}
  >
    <Text
      style={
        [
          {
            "backgroundColor": "#fff",
            "color": "#2E50D4",
            "fontFamily": "Nunito-Bold",
            "fontSize": 14,
            "left": 16,
            "padding": 4,
            "position": "absolute",
            "top": -14,
          },
          undefined,
        ]
      }
    >
      label
    </Text>
    <View
      style={
        [
          {
            "flex": 1,
            "flexDirection": "row",
          },
          undefined,
        ]
      }
    >
      <TextInput
        multiline={false}
        placeholderTextColor="#B3B3B3"
        style={
          [
            {
              "color": "#333333",
              "flex": 1,
              "fontFamily": "Nunito-SemiBold",
              "fontSize": 16,
              "padding": 0,
            },
            undefined,
          ]
        }
        testID="input-text"
        textAlignVertical="center"
      />
    </View>
  </View>
</View>
`;
